hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java

```diff
@@ -770,6 +771,33 @@ public void testLookupKeyWithLocation() throws IOException {
         .getLocationList().get(0).getPipeline().getClosestNode());
   }
 
+  @Test
+  public void testListStatusWithTableCache() throws Exception {
+    // Inspired by TestOmMetadataManager#testListKeys
+    String prefixKeyInDB = "key-d";
+    String prefixKeyInCache = "key-c";
+
+    for (int i = 1; i <= 100; i++) {
+      if (i % 2 == 0) {  // Add to DB
+        TestOMRequestUtils.addKeyToTable(false,
+            VOLUME_NAME, BUCKET_NAME, prefixKeyInDB + i,
+            1000L, HddsProtos.ReplicationType.RATIS,
+            HddsProtos.ReplicationFactor.ONE, metadataManager);
+      } else {  // Add to TableCache
```

Comment:
Can we add a few combinations of tests similar to listKeys.
1. Where some entries marked for delete.
2. With a pagination test.

Reply:
I just added new tests in a new commit. But there are some failures. I'm looking into it.